src/base/QXmppDataForm.h

```diff
@@ -120,10 +124,50 @@ class QXMPP_EXPORT QXmppDataForm
         QVariant value() const;
         void setValue(const QVariant &value);
 
+        void parse(const QDomElement &element);
+        void toXml(QXmlStreamWriter *writer) const;
     private:
         QSharedDataPointer<QXmppDataFormFieldPrivate> d;
     };
 
+    class QXMPP_EXPORT Item
+    {
+    public:
+        Item(const QXmppDataForm::Item &other);
+        Item();
+        ~Item();
+
+        QXmppDataForm::Item& operator=(const QXmppDataForm::Item &other);
+
+        QList<Field> fields() const;
+        QList<Field> &fields();
+        void setFields(const QList<QXmppDataForm::Field> &fields);
+
+        void parse(const QDomElement &element);
+        void toXml(QXmlStreamWriter *writer) const;
+    private:
+        QSharedDataPointer<QXmppDataFormItemPrivate> d;
+    };
+
+    class QXMPP_EXPORT Reported
```

Comment:
I am not entirely sure I understand the purpose of the Item / Reported classes. Unless I am mistaken, they both seem to describe a list/vector of Fields?

Reply:
there is a little different xml tag name and element between reported, item

if you would like it to use only item, it is possible to, but there will be some specific logic for this kind purpose.

like below
```
 "<x xmlns=\"jabber:x:data\" type=\"result\">"
            "<reported>"
                "<field type=\"jid-single\" label=\"JID\" var=\"jid\"/>"
                "<field type=\"text-single\" label=\"Username\" var=\"Username\"/>"
                "<field type=\"text-single\" label=\"Name\" var=\"Name\"/>"
                "<field type=\"text-single\" label=\"Email\" var=\"Email\"/>"
            "</reported>"
            "<item>"
                "<field type=\"text-single\" var=\"jid\">"
                    "<value>Zam@im-BestTaiwan.com.tw</value>"
                "</field>"
                "<field type=\"text-single\" var=\"Username\">"
                    "<value>Zam</value>"
                "</field>"
                "<field type=\"text-single\" var=\"Name\">"
                    "<value>Zam</value>"
                "</field>"
                "<field type=\"text-single\" var=\"Email\">"
                    "<value>Zam@BestTaiwan.com.tw</value>"
                "</field>"
            "</item>"
        "</x>"
```

Reply:
@jlaine 
if you are having any considering or suggestion, please shoot to me.
it would be great to have that.

Reply:
OK so "reported" has "label" but no "value", right? Let's keep the classes separate then.